apps/extension/src/core/domains/mnemonics/store.ts

```diff
@@ -0,0 +1,130 @@
+import { StorageProvider } from "@core/libs/Store"
+import { log } from "@core/log"
+import { decrypt, encrypt } from "@metamask/browser-passworder"
+import { assert, isObject } from "@polkadot/util"
+import { nanoid } from "nanoid"
+import { Err, Ok, Result } from "ts-results"
+
+import { LegacySeedObj, legacyDecryptSeed } from "./legacy/helpers"
+
+const storageKey = "seeds"
+
+export enum SOURCES {
+  Imported = "imported",
+  Generated = "generated",
+  Legacy = "legacy",
+  Vault = "vault",
+}
+
+export type SeedPhraseData = {
+  id: string
+  name: string
+  cipher?: string
+  source: SOURCES
```

Comment:
Removed 'vault' in https://github.com/TalismanSociety/talisman/pull/1050. I think we should keep the `source` field, since it might be useful to show whether a mnemonic is imported or created at some point.